packages/runtime/container-runtime/src/blobManager.ts

```diff
@@ -77,20 +74,24 @@ export class BlobManager {
             `${BlobManager.basePath}/${response.id}`,
             this.routeContext,
             async () => this.getStorage().readBlob(response.id),
-            () => this.attachBlobCallback(response.id),
         );
 
         // Note - server will de-dup blobs, so we might get existing blobId!
         if (!this.blobIds.has(response.id)) {
```

Comment:
For consistency, you likely want to call handle.attachGraph() in ELSE clause.
That said, we have a race condition here if we do so, if two same images gets uploaded at the same time.
Can BlobHandle instead of having attached property check with this.pendingBlobIds to determine if it is attached or not?